src/nvim/quickfix.c

```diff
@@ -3983,17 +4000,54 @@ static int qf_buf_add_line(buf_T *buf, linenr_T lnum, const qfline_T *qfp,
   return OK;
 }
 
+static list_T *call_qftf_func(qf_list_T *qfl,
+                              int qf_winid,
+                              long start_idx,
+                              long end_idx)
+{
+  char_u *qftf = p_qftf;
+  list_T *qftf_list = NULL;
+
+  // If 'quickfixtextfunc' is set, then use the user-supplied function to get
+  // the text to display. Use the local value of 'quickfixtextfunc' if it is
+  // set.
+  if (qfl->qf_qftf != NULL) {
+    qftf = qfl->qf_qftf;
+  }
+  if (qftf != NULL && *qftf != NUL) {
+    typval_T args[1];
+    dict_T *const dict = tv_dict_alloc();
```

Comment:
Vim locked this.
Does the code break if it's locked?

Reply:
It's necessary because that's what in the vim patch. If the PR you're referring to diverged from the vim patch without explanation (ie. not locking the dictionary that should be locked), then that is the fault of the porter. Stick to the vim patch as much as possible and document any necessary divergences in the commit message.

If existing code or test fails, then an unmerged vim patch is likely missing and it should be ported in the same PR if possible.